src/pybind/mgr/dashboard/module.py

```diff
@@ -751,6 +752,133 @@ def get_perf_schema(self, **args):
+                    }
+
+                result = []
+                sync_data = []
+                osd_count = len(data)
+
+                ref_osd = None
+                epoch_set = {}
+                for osd_id in data:
+                    epoch = str(data[osd_id][0][0])
+                    if epoch not in epoch_set:
+                        epoch_set[epoch] = []
+                    epoch_set[epoch].append(osd_id)
+                max_sync = 0
+                for epoch in epoch_set:
+                    sync_count = len(epoch_set[epoch])
+                    if sync_count > max_sync:
+                        ref_osd = epoch_set[epoch][0]
+                        max_sync = sync_count
+
+                for i in range(0,len(data[ref_osd])):
+
+                      epoch = data[ref_osd][i][0]
+                      values = []
+                      outsync_osds = []
+
+                      for osd_id in data:
+                            if data[osd_id][i][0] != epoch:
+                                outsync_osds.append(osd_id)
+                            values.append(data[osd_id][i][1])
+
+                      result.append(
+                            [epoch, values])
+                      counter_stats_data["osds_out_sync"].append(
+                            [epoch, outsync_osds])
+
+                for tseq in result:
+                    
+                    length = len(tseq[1])
+                    avg = sum(tseq[1])/length
+                    counter_stats_data["avg"].append([tseq[0], avg])
+                    diff = [(x-avg)**2 for x in tseq[1]]
+                    var = sum(diff)/length
+                    stdev = math.sqrt(var)
+                    counter_stats_data["stdev"].append([tseq[0], stdev])
+                    counter_stats_data["stdevs"]["plus_onehalf"].append([tseq[0], avg+(1.5*stdev)])
+                    counter_stats_data["stdevs"]["plus_half"].append([tseq[0], avg+(0.5*stdev)])
+                    counter_stats_data["stdevs"]["minus_half"].append([tseq[0], avg-(0.5*stdev)])
+                    counter_stats_data["stdevs"]["minus_onehalf"].append([tseq[0], avg-(1.5*stdev)])
+                    counter_stats_data["max"].append([tseq[0], max(tseq[1])])
+                    counter_stats_data["min"].append([tseq[0], min(tseq[1])])
+
+                return counter_stats_data
+
+            @cherrypy.expose
+            def perf_graphs_prototypes(self, path):
+                template = env.get_template("perf_graphs_prototypes.html")
+                toplevel_data = self._toplevel_data()
+
+                content_data = self.get_counter_stats(path)
+
+                return template.render(
+                    ceph_version=global_instance().version,
+                    toplevel_data=json.dumps(toplevel_data, indent=2),
+                    content_data=json.dumps(content_data, indent=2),
+                )
+
+            @cherrypy.expose
+            @cherrypy.tools.json_out()
```

Comment:
again, why shall we expose this method?

Reply:
Again, This function is no longer there in the final commit. I would like your opinion on squashing all the changes related to graphs and TS in a single commit.